internal/manager/config.go

```diff
@@ -68,8 +69,8 @@ type Config struct {
 	GatewayAPIControllerName string
 
 	// Ingress status
-	PublishServiceUDP       types.NamespacedName
-	PublishService          types.NamespacedName
+	PublishServiceUDP       mo.Option[types.NamespacedName]
+	PublishService          mo.Option[types.NamespacedName]
```

Comment:
One thing I'd like to call out is that this is already becoming unreadable for the end user.

```
      --kong-admin-svc Validatedmo.Option[k8s.io/apimachinery/pkg/types.NamespacedName]        Kong Admin API Service namespaced name in "namespace/name" format, to use for Kong Gateway service discovery.
      --publish-service-udp Validatedmo.Option[k8s.io/apimachinery/pkg/types.NamespacedName]   Service fronting UDP routing resources in "namespace/name" format. The controller will update UDP route status information with this Service's endpoints. If omitted, the same Service will be used for both TCP and UDP routes.
```

Even extracting to a separate type with a "nice" name:

```
      --publish-service Validatedmanager.OptionalNamespacedName                                Service fronting Ingress resources in "namespace/name" format. The controller will update Ingress status information with this Service's endpoints.
```

What I suggest is that we create a type and allow `Type()` overrides in https://github.com/Kong/kubernetes-ingress-controller/blob/95c97ce2ff992a7d29d8294d14a8da0173bea2de/internal/manager/validated.go#L47-L50 so that the type can have its own name shown in `--help` output.

I don't see a reason for including `Validated` prefix for this. Something like `NamespacedName` would be short and to the point for me. The details of the usage can be put in the flag description.